pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -220,5 +222,29 @@ func (r *ReconcileJaeger) apply(jaeger v1.Jaeger, str strategy.S) error {
 		}
 	}
 
+	if err := r.scanDeployments(jaeger); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (r *ReconcileJaeger) scanDeployments(jaeger v1.Jaeger) error {
```

Comment:
What does the func do? Does it scan for something or does it add sidecars?

Reply:
The name is wrong, not intuitive at all, sorry

It suppose to add the sidecar for deployments that require that. But I think I found a better way for do this.